includes/discovery/functions.inc.php

```diff
@@ -661,55 +664,61 @@ function discover_process_ipv6(&$valid, $ifIndex, $ipv6_address, $ipv6_prefixlen
     $ipv6_network = $ipv6->getNetwork($ipv6_prefixlen);
     $ipv6_compressed = $ipv6->compressed();
 
-    if (dbFetchCell('SELECT COUNT(*) FROM `ports` WHERE device_id = ? AND `ifIndex` = ?', [$device['device_id'], $ifIndex]) != '0' && $ipv6_prefixlen > '0' && $ipv6_prefixlen < '129' && $ipv6_compressed != '::1') {
-        $port_id = dbFetchCell('SELECT port_id FROM `ports` WHERE device_id = ? AND ifIndex = ?', [$device['device_id'], $ifIndex]);
+    $port_id = Port::where([
+        ['device_id', $device['device_id']],
+        ['ifIndex', $ifIndex],
+    ])->value('port_id');
 
-        if (is_numeric($port_id)) {
-            if (dbFetchCell('SELECT COUNT(*) FROM `ipv6_networks` WHERE `ipv6_network` = ?', [$ipv6_network]) < '1') {
-                dbInsert(['ipv6_network' => $ipv6_network, 'context_name' => $context_name], 'ipv6_networks');
-                echo 'N';
-            } else {
-                //Update Context
-                dbUpdate(['context_name' => $device['context_name']], 'ipv6_networks', '`ipv6_network` = ?', [$ipv6_network]);
-                echo 'n';
-            }
+    if ($port_id && $ipv6_prefixlen > '0' && $ipv6_prefixlen < '129' && $ipv6_compressed != '::1') {
+        d_echo('IPV6: Found port id: ' . $port_id);
 
-            if ($context_name == null) {
-                $ipv6_network_id = dbFetchCell('SELECT `ipv6_network_id` FROM `ipv6_networks` WHERE `ipv6_network` = ? AND `context_name` IS NULL', [$ipv6_network]);
-            } else {
-                $ipv6_network_id = dbFetchCell('SELECT `ipv6_network_id` FROM `ipv6_networks` WHERE `ipv6_network` = ? AND `context_name` = ?', [$ipv6_network, $context_name]);
+        $ipv6netDB = Ipv6Network::firstOrNew([
+            'ipv6_network' => $ipv6_network,
+        ], [
+            'context_name' => $context_name,
+        ]);
+
+        //check if DB data match and check for DB NULL
+        if ($ipv6netDB->context_name != $context_name || $ipv6netDB->context_name == null) {
```

Comment:
Models support an "isDirty()" method, this will check if the model has changed fields so you can print out the updated message.

Reply:
My preferred method is updateOrCreate and use a listener to see when the devices is updated or created.

Reply:
modified as you suggested
learned something new today, yes it is much simpler this way